File: src/types/cook-session.ts
export type CookStatus = 'idle' | 'running' | 'paused' | 'completed' | 'failed'

export interface CookSession {
  id: string
  taskId: string
  worktreeId?: string
  ccsProfile: string
  sessionLogPath?: string
  status: CookStatus
  startedAt?: string
  finishedAt?: string
  createdAt: string
  updatedAt: string
}

File: src-tauri/src/models/cook_session.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum CookStatus {
    Idle,
    Running,
    Paused,
    Completed,
    Failed,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "camelCase")]
pub struct CookSession {
    pub id: String,
    pub task_id: String,
    pub worktree_id: Option<String>,
    pub ccs_profile: String,
    pub session_log_path: Option<String>,
    pub status: CookStatus,
    pub started_at: Option<String>,
    pub finished_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}
